docs/Software Engineering Blog/posts/spring-boot-3-1-docker-compose-support-intuitive-consistent-develop-run-tests-containers.md

```diff
@@ -0,0 +1,229 @@
+follow the starting guide could easily run the app with the dependency container together in one command.
+
+just add the dependency
+
+```groovy
+developmentOnly 'org.springframework.boot:spring-boot-docker-compose'
+```
+
+then provide a `compose.yaml` file in the root folder can do the trick. If the compose.yaml has a different name or in a
+different folder, `spring.docker.compose.file` can solve the problem, refer
+to [Using a Specific Compose File](https://docs.spring.io/spring-boot/reference/features/dev-services.html#features.dev-services.docker-compose.specific-file).
+
+![img.png](resources/sb-docker/start_with_docker_console.png)
+
+![start_with_docker_podman.png](resources/sb-docker/start_with_docker_podman.png)
+
+From now on, every project member can run the app without manually starting the dependency component container
+cluster.
+
+## Little More Setup to Make It Better
+
+To enhance the development and testing experience, consider the following improvements.
+
+### For Development
+
+#### Docker Profile Connected to Spring Profile:
+
+Since the practice of spring profiles is usually used as a dependency management method. For example, we add a
+profile `redis` in `spring.profiles.include`, to include `application-redis.yaml`, which contains the related config for
+using redis.
+
+So we want to link docker compose profiles to Spring profiles. In this way, developers can ensure that the correct
+configuration is used and the corresponding container is up also.
+
+To do that, we need some modification:
+
+1. use comma separate format for `spring.profiles.include`
+
+  ```yaml title="application.yaml"
+  spring:
+    application:
+      name: order-command-side
+    profiles:
+      include: "observe, cluster, mock, kafka, kafka-producer, redis, idempotency, docker-support"
+  ```
+
+2. add profile linking property 'spring.docker.compose.profiles.active'
+
+  ```yaml title="application-docker-support.yaml"
+  spring:
+    docker:
+      compose:
+        enabled: true
+        profiles:
+          active: ${spring.profiles.include}
+  ```
+
+3. add profile for each service in compose.yaml
+
+  ```yaml title="compose.yaml"
+  version: "3"
+
+  services:
+    kafka:
+      image: 'bitnami/kafka:3.7'
+      profiles: ["kafka", "all"]
+      ports:
+        - '9092:9092'
+      environment:
+        - KAFKA_CFG_NODE_ID=0
+        - KAFKA_CFG_PROCESS_ROLES=controller,broker
+        - KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093
+        - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
+        - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://${KAFKA_HOST:-localhost}:9092
+        - KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=0@kafka:9093
+        - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
+
+    redis:
+      image: 'redis:7.2'
+      profiles: ["redis", "all"]
+      ports:
+        - '6379:6379'
+  ```
+
+#### Ignore when Building in Image:
```

Comment:
⚠️ **[vale]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Google.Headings] 'Ignore when Building in Image:' should use sentence-style capitalization.